Dockerfile

```diff
@@ -1,29 +1,9 @@
 FROM ocaml/opam:debian-12-ocaml-4.14@sha256:06d58a5cb1ab7875e8d94848102be43f6492e95320e8e9a9ecb9167654d0ee3f AS build
 RUN sudo apt-get update && sudo apt-get install libffi-dev libev-dev m4 pkg-config libsqlite3-dev libgmp-dev libssl-dev capnproto graphviz -y --no-install-recommends
-RUN cd ~/opam-repository && git fetch -q origin master && git reset --hard 6d3d7021d944058cc7531905e065fe2cd72f01e8 && opam update
-COPY --chown=opam \
-	ocurrent/current_docker.opam \
-	ocurrent/current_github.opam \
-	ocurrent/current_git.opam \
-	ocurrent/current.opam \
-	ocurrent/current_rpc.opam \
-	ocurrent/current_slack.opam \
-	ocurrent/current_ssh.opam \
-	ocurrent/current_web.opam \
-	/src/ocurrent/
-COPY --chown=opam \
-        ocluster/*.opam \
-        /src/ocluster/
+RUN cd ~/opam-repository && git fetch -q origin master && git reset --hard 99bc90ff813af4d02cb0627f6b3e5a8c84e2e04a && opam update
 WORKDIR /src
-RUN opam pin add -yn current_docker.dev "./ocurrent" && \
-    opam pin add -yn current_github.dev "./ocurrent" && \
-    opam pin add -yn current_git.dev "./ocurrent" && \
-    opam pin add -yn current.dev "./ocurrent" && \
-    opam pin add -yn current_rpc.dev "./ocurrent" && \
-    opam pin add -yn current_slack.dev "./ocurrent" && \
-    opam pin add -yn current_ssh.dev "./ocurrent" && \
-    opam pin add -yn current_web.dev "./ocurrent" && \
-    opam pin add -yn ocluster-api.dev "./ocluster"
+# See https://github.com/ocurrent/ocaml-docs-ci/pull/177#issuecomment-2445338172
+RUN sudo chown opam:opam $(pwd)
```

Comment:
Are you sure that this is needed?  It generally isn't.  Usually the trick is to set `WORKDIR` after the a `COPY`: e.g.

```
COPY --chown=opam deployer.opam /src/
WORDIR /src
```

If you look at the original `Dockerfile`, you'll see that this was the case but by removing the `COPY` on line 4 and 14 it has changed.  This also happened in https://github.com/ocurrent/ocaml-docs-ci/pull/177

See https://github.com/ocurrent/opam-repo-ci/blob/master/Dockerfile.

Reply:
Oh that is helpful. All my searching on this irritating matter did not lead me to learn that it is order dependent in that way.

I'll swap the order and remove the chown. Thanks.